kv/txn_coord_sender.go

```diff
@@ -758,28 +756,26 @@ func (tc *TxnCoordSender) heartbeat(id string) {
 					Reply: reply,
 				}
 
-				task := tc.stopper.StartTask()
-				if !task.Ok() {
+				if !tc.stopper.RunTask(func() {
```

Comment:
> removing the return values will require scrutiny of each place where they were used

Agreed.

> so it should get its own PR.

This change already introduces ugliness to work around the slight reduction in expressivity that arises (e.g. https://github.com/cockroachdb/cockroach/pull/1640/files#diff-dae9f8a58ca2ec436440b12989c1982dR1440). Don't you think it's worth it to flesh this out before heading in this direction?

Or do you disagree that leaving it as presented here feels half-baked?